census/vanilla-tf/trainer/task.py

```diff
@@ -44,228 +43,149 @@
-
-  with session:
-    coord = tf.train.Coordinator()
-    threads = tf.train.start_queue_runners(coord=coord, sess=session)
-
-    while step < max_steps:
-      session.run(
-          train_step,
-          feed_dict={
-              inputs: session.run(train_inp),
-              labels: session.run(train_label)
-          }
-      )
-      step = tf.train.global_step(session, global_step)
-      if step % 10 == 0:
-        accuracy = evaluate_accuracy(session, eval_step, inputs, labels,
-                                   eval_inp, eval_label)
-        print('[{}/{}]: Step number {} of {} done, Accuracy {:.2f}%'.format(
-            job_name, job_id, step, max_steps, accuracy))
-
-    coord.request_stop()
-    coord.join(threads)
-
-  return train_step
-
-
-#
-# Single and Distributed training functions.
-#
-def training_single(job_dir, max_steps, train_eval_tensor):
-  """Perform single node training."""
+def run(target,
+        is_chief,
+        max_steps,
+        train_data_paths,
+        eval_data_paths,
+        job_dir,
+        eval_every=100,
+        eval_steps=10,
+        learning_rate=0.1,
+        num_epochs=None,
+        batch_size=40):
 
-  inputs, labels = get_placeholders()
-
-  graph = make_graph(inputs, labels)
-  init = tf.global_variables_initializer()
-
-  session = tf.train.MonitoredTrainingSession(checkpoint_dir=job_dir,
-                                              save_checkpoint_secs=20,
-                                              save_summaries_steps=50)
-  session.run(init)
-
-  training_steps(session, graph, inputs, labels, max_steps, train_eval_tensor)
-
-def training_distributed(job_dir, max_steps, train_eval_tensor):
-  """Perform distributed training."""
-
-  # Parse the TF_CONFIG to create cluster spec and job name
-  # We are doing this manually here to demonstrate
-  # See RunConfig abstraction here https://goo.gl/f4BQMo
-  cluster_spec, job_name, task_index = parse_tf_config()
-
-  # Create and start a server
-  server = tf.train.Server(cluster_spec,
-                           job_name=job_name,
-                           task_index=task_index)
-
-  is_chief = (job_name == 'master')
+  """Run the training steps and calculate accuracy every 10 steps."""
 
-  if job_name == 'ps':
-    server.join()
-  elif job_name in ['master', 'worker']:
+  training_eval_graph = tf.Graph()
+  with training_eval_graph.as_default():
     with tf.device(tf.train.replica_device_setter()):
-      inputs, labels = get_placeholders()
-      graph = make_graph(inputs, labels)
-
-    init = tf.global_variables_initializer()
-
-    with tf.train.MonitoredTrainingSession(master=server.target,
+      mode = tf.placeholder(shape=[], dtype=tf.string)
+      eval_features, eval_label = input_fn(
+          eval_data_paths, shuffle=False, batch_size=batch_size)
+      train_features, train_label = input_fn(
+          train_data_paths, shuffle=True, num_epochs=num_epochs, batch_size=batch_size)
```

Comment:
In this logic between lines 137-154 can we not create two graphs one for training and one for eval directly vs doing this cond thing?